lib/src/components/NxProgressBar/NxProgressBar.tsx

```diff
@@ -0,0 +1,123 @@
+ */
+import { faExclamationCircle, faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import classnames from 'classnames';
+import React, { forwardRef } from 'react';
+
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { Props, propTypes } from './types';
+
+import './NxProgressBar.scss';
+
+const NxProgressBar = forwardRef<HTMLProgressElement, Props>(
+    function NxProgressBar(props, ref) {
+      const {
+        className,
+        hasError,
+        inlineCounter,
+        label,
+        labelError,
+        labelSuccess,
+        max: maxProp,
+        showCounter: showCounterProp,
+        value,
+        variant,
+        ...otherAttributes
+      } = props;
+
+      const showCounter = showCounterProp ?? true;
+      const max = maxProp ?? 100;
+      const percentage = Math.round(value / max * 100);
+
+      let labelText = label;
+
+      const renderCounter = () => {
+        if (!showCounter || variant === 'inline') {
+          return null;
+        }
+
+        return <span className="nx-counter nx-progress-bar__counter">{`${hasError ? 0 : percentage}%`}</span>;
+      };
+
+      const renderLabel = () => {
+        if (variant === 'inline' || variant === 'small') {
+          return null;
+        }
+
+        if (hasError) {
+          if (labelError) {
+            labelText = labelError;
+            return (
+              <>
+                <NxFontAwesomeIcon icon={faExclamationCircle} />
+                <span className="nx-progress-bar__label-text">{labelText}</span>
+              </>
+            );
+          }
+          else {
+            console.warn('labelError is required when hasError is true.');
+          }
+        }
+
+        if (percentage === 100) {
+          labelText = labelSuccess || labelText;
+          return (
+            <>
+              <NxFontAwesomeIcon icon={faCheckCircle} />
+              <span className="nx-progress-bar__label-text">{labelText}</span>
+            </>
+          );
+        }
+
+        return <span className="nx-progress-bar__label-text">{labelText}</span>;
+      };
+
+      const renderCounterAndLabel = () => {
+        if (variant === 'inline') {
+          return null;
+        }
+
+        return (
+          <span className="nx-progress-bar__counter-and-label">
+            {renderCounter()}
+            {inlineCounter ? null : renderLabel()}
+          </span>
+        );
+      };
+
+      const classes = classnames('nx-progress-bar', {
+        'nx-progress-bar--inline': variant === 'inline',
+        'nx-progress-bar--small': variant === 'small',
+        'nx-progress-bar--full': variant === 'full',
+        'nx-progress-bar--inline-counter': inlineCounter,
+        'nx-progress-bar--error': hasError,
+        'nx-progress-bar--success': percentage === 100
+      }, className);
+
+      const WrapperElement = variant === 'inline' ? 'span' : 'label';
```

Comment:
This needs to be a `label` always.  You can set it to inline in CSS

Reply:
> This needs to be a `label` always. You can set it to inline in CSS

Fixed here: 90ff9ad9be471cf52e92754ad72052e46b897545